backups/cognito/backup_users.py

```diff
@@ -0,0 +1,76 @@
+import boto3
+import csv
+from datetime import datetime
+from botocore.exceptions import ClientError
+import os
+
+client = boto3.client('cognito-idp')
+s3_client = boto3.client('s3')
+
+environment='production'
+user_pool_name = 'biomage-user-pool-case-insensitive-{}'.format(environment)
+backup_bucket = 'biomage-backups-{}'.format(environment)
+
+def backup_users():
+    try:
+        all_users = get_all_users()
+        num_users = len(all_users)
+    except Exception as e:
```

Comment:
* I recommend declaring all_users outside the try-block  or it might be out-of-scope later
* do not use generic exceptions,  use the specific one you want to catch.
* wrap only `all_users = get_all_users()` in the try block

Reply:
It wont be out of scope because if  the try fails, the script stops

Reply:
Also there are many things that can go wrong I want to stop the script every time so can i not use the generic exception?